src/onINP.ts

```diff
@@ -84,28 +158,37 @@ const processEntry = (entry: PerformanceEventTiming) => {
     longestInteractionList.length < MAX_INTERACTIONS_TO_CONSIDER ||
     entry.duration > minLongestInteraction.latency
   ) {
+    const entriesGroup = pendingEntriesGroupMap[renderTime];
+
     // If the interaction already exists, update it. Otherwise create one.
     if (existingInteraction) {
-      existingInteraction.entries.push(entry);
-      existingInteraction.latency = Math.max(
-        existingInteraction.latency,
-        entry.duration,
-      );
+      if (entry.duration > existingInteraction.latency) {
+        existingInteraction.latency = entry.duration;
+        if (existingInteraction.renderTime !== renderTime) {
+          existingInteraction.renderTime = renderTime;
+          existingInteraction.entries = entriesGroup!.entries;
+        }
+      }
     } else {
       const interaction = {
         id: entry.interactionId!,
         latency: entry.duration,
-        entries: [entry],
+        entries: entriesGroup!.entries,
+        renderTime,
       };
       longestInteractionMap[interaction.id] = interaction;
       longestInteractionList.push(interaction);
     }
 
     // Sort the entries by latency (descending) and keep only the top ten.
     longestInteractionList.sort((a, b) => b.latency - a.latency);
-    longestInteractionList.splice(MAX_INTERACTIONS_TO_CONSIDER).forEach((i) => {
-      delete longestInteractionMap[i.id];
-    });
+    if (longestInteractionList.length > MAX_INTERACTIONS_TO_CONSIDER) {
```

Comment:
Is this guard actually needed or just making it more documented?

Reply:
I guess not technically, but it seemed more efficient to have an if statement than unconditionally calling `splice()` on an array that may not need it.